gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/tasks/snapshots/transform/FindCustomPreviewClassVisitor.kt

```diff
@@ -30,10 +30,83 @@ class FindCustomPreviewClassVisitor(
   override fun visitAnnotation(descriptor: String?, visible: Boolean): AnnotationVisitor? {
     if (descriptor == EMERGE_APP_STORE_SNAPSHOT) {
       currentAnnotation.isAppStoreSnapshot = true
-    } else if (customPreviewAnnotations.containsKey(descriptor)) {
+    } else if (descriptor == PREVIEW_ANNOTATION_DESC) {
       currentAnnotation.hasPreviewAnnotation = true
+
+      // Capture preview parameters
+      return object : AnnotationVisitor(api) {
+        override fun visit(name: String?, value: Any?) {
+          capturePreviewParameter(currentAnnotation, name, value)
+          super.visit(name, value)
+        }
+      }
+    } else if (descriptor == PREVIEW_LIGHT_DARK_ANNOTATION_DESC) {
```

Comment:
Not quite sure I like this approach of hardcoding the states for these. I'm assuming the current multipreview parsing is not picking these up from their definitions in the tooling library([source](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/ui/ui-tooling-preview/src/androidMain/kotlin/androidx/compose/ui/tooling/preview/MultiPreviews.android.kt;l=75?q=PreviewLightDark&sq=))?

Reply:
That’s correct. Unfortunately, we don’t have access to the tooling library since we filter out third party libraries. So my idea was to hard code them. See `PreviewConstants`.

The logic around them was in `AnnotatedMethodVisitor#createComposePreviewConfigs` so this was duplicated hard-coding of that library. I turned this PR in to a big refactor in order to reduce that duplication as well as other pieces of duplication in all this logic.